pkg/api/security/security.go

```diff
@@ -117,7 +116,7 @@ func GetAuthTokenFilepath(config configModel.Reader) string {
 	if config.GetString("auth_token_file_path") != "" {
 		return config.GetString("auth_token_file_path")
 	}
-	return filepath.Join(filepath.Dir(config.ConfigFileUsed()), authTokenName)
+	return filepath.Join(config.GetString("run_path"), authTokenName)
```

Comment:
@vboulineau While it makes sense to not use `/etc` here, I don't think we can be sure that `/opt/datadog-agent/run` as-is will provide adequate security (only accessible by the agent user) to the `auth_token`. This also puts into question `config.GetString("run_path")`since it is configurable and could be changed to a path that does not provide enough assurances on the access. I'd suggest that this goes maybe to SIT team for eval as it could have unforeseen impacts and I'll ping ASC to take a look as well.

Reply:
I checked the security bits before opening and the PR and I believe it's the same as before:
- Currently we take directory of `datadog.yaml`, which could also be anywhere.
- `/etc/datadog-agent` and `/opt/datadog-agent/run` are both OOTB `755` `dd-agent:dd-agent`.
- `auth_token` file is written `600` and is owned `dd-agent:dd-agent`.

Happy to dive deeper if necessary.

Reply:
Fair enough. Are we sure about this setup on all relevant permutations though (Windows, Darwin, etc). So far, the main concerns from the team outside of my listed ones are making sure we don't break something accidentally (CLI, some user tool) and that public docs are updated properly.

Reply:
> Are we sure about this setup on all relevant permutations though (Windows, Darwin, etc).
The path is already used for other private data (like remote config DB) and permission on this folder looks good to me on Darwin and Windows. There are also E2E tests running CLI commands.

> we don't break something accidentally (CLI, some user tool) and that public docs are updated properly.
I checked the public documentation and I did not find any mention of the current path, we cannot rule out that someone built a tool somewhere using this path but we don't document it.

On our own CLI, that should work as the `agent` (and friends) binaries are the same versions.